pkg/tfbridge/typechecker/typechecker_test.go

```diff
@@ -1631,6 +1631,11 @@ func TestValidateConfigType(t *testing.T) {
 				})),
 			}),
 		},
+		{
```

Comment:
The test isn't great. It exercises the changed code path workaround which is good but it does assert something that will need to be retracted once the root cause is fixed. I think if we add a todo to the root cause ticket in here that'll help us remember later on. Thank you!